src/main/java/core/basesyntax/service/impl/SupplyTransactionHandler.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Database;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.service.TransactionHandler;
+import java.math.BigInteger;
+
+public class SupplyTransactionHandler implements TransactionHandler {
+    @Override
+    public void handle(Database database, FruitType fruitType, BigInteger quantity) {
```

Similarly, `fruitType` is not validated. A null `fruitType` would insert a null key into storage (or cause unexpected behavior later). Validate `fruitType` at the start of the method and fail fast with an informative message (for example: `Objects.requireNonNull(fruitType, "FruitType must not be null in SupplyTransactionHandler.handle")`). This addresses the checklist requirement to handle null parameters.